File: cloudsearchifyDocuments.py
#!/usr/bin/env python

# script to set up apache development mode on Mac OSX
# all apache files need writeable permissions for this script to run
import json, uuid
documentsFile = '/Users/srice/Desktop/allHealthPlans.json'
destination = '/Users/srice/Desktop/csAllHealthPlans.json'

try:      
  csAllHealthPlans = []
  # with open('/usr/local/bin/projectBranches.json') as jsonFile:    
  with open(documentsFile) as jsonFile:    
     allHealthPlans = json.load(jsonFile) 

     for healthPlan in allHealthPlans:
       for key,value in healthPlan.iteritems():
         if type(value) is dict:
           for key2,value2 in value.iteritems():
             healthPlan[key] = value2

       csHealhPlan = {
         "type": "add",
         "id": str(uuid.uuid4())
       }

       csHealhPlan["fields"] = healthPlan

       csAllHealthPlans.append(csHealhPlan);

  with open(destination, 'w') as outfile:
    json.dump(csAllHealthPlans, outfile)
    print "Writing file to file at /Users/srice/Desktop/csAllHealthPlans.json"

except IOError as (errno,strerror):
    print "I/O error({0}): {1}".format(errno, strerror)

File: cloudsearchifyDocuments.js
// This script expects an array of json objects.
// It 1) adds id and type fields to each object, since Cloudsearch expects all JSON objects to have these fields, and
// 2) converts any value fields that are of type "object" to type string, since Cloudsearch cannot index json objects as field values.
var uuid = require('node-uuid');

exports.cloudsearchifyDocuments = function(allHealthPlans) {
  var csAllHealthPlans = [];
  for(var i = 0; i < allHealthPlans.length; i++) {
    var healthPlan = allHealthPlans[i];
    for(var key in healthPlan) {
      if(healthPlan.hasOwnProperty(key) && typeof healthPlan[key] === "object") {
        for(var key2 in healthPlan[key]) {
          if(healthPlan[key].hasOwnProperty(key2)) {
            healthPlan[key] = healthPlan[key][key2];
          }
        }
      }
    }

    var csHealhPlan = {
      "type": "add",
      "id": uuid.v4()
    };

    csHealhPlan["fields"] = healthPlan;
    csAllHealthPlans.push(csHealhPlan);
  }

  return JSON.stringify(csAllHealthPlans, null, 4);
};
